Public/Src/Utilities/Configuration/ISandboxConfiguration.cs

```diff
@@ -220,5 +220,15 @@ public interface ISandboxConfiguration
         /// This is used mainly for testing.
         /// </remarks>
         AbsolutePath RedirectedTempFolderRootForVmExecution { get; }
+
+        /// <summary>
+        /// Retries process whose exit code or its children's exit code is 0xDEAD.
+        /// </summary>
+        /// <remarks>
+        /// Process can nondeterministically exit with 0xDEAD exit code in CB. This is the exit code
+        /// returned by Azure Watson dump after catching the process crash. The root cause
+        /// of the crash is unknown, but the primary suspect is the way Detours handle NtClose.
+        /// </remarks>
+        bool RetryOnDeadExitCode { get; }

```

Comment:
What's the scope of this? Is it just going to be for QB? Just for the specific pip that's hitting the detours break?

Reply:
Traditionally we've treated a compiler or other tool crash as any other build failure. Having a distinct exit code for crashes in AzureWatson enabled environments is an interesting angle. WDG has asked about having the ability to treat tool crashes different in reporting. No action from the BuildXL, just sharing some thoughts

Reply:
For QuickBuild case, there's a queue and there are known targets in that queue that fail 95% of time with this exit code. I try to isolate the problem, and it seems to root at our detoured `NtClose`. 

Office also experiences this issue. But so far, from the Azure Watson's dumps, they cannot make definite conclusion what may have cause it. For Office, it's less often, and its non-deterministic nature makes it difficult to perform investigation.

---
In reply to: [289950639](https://github.com/microsoft/BuildXL/pull/401#discussion_r289950639) [](ancestors = 289950639)

Reply:
" to treat tool crashes different in reporting" -- how different in reporting do they want? Or, what kind of reporting do they want?

---
In reply to: [289961049](https://github.com/microsoft/BuildXL/pull/401#discussion_r289961049) [](ancestors = 289961049)